### ISSUE TITLE
Cop::Style::SpaceInsideHashLiteralBraces auto-correct mangles a hash with a hash key

### ISSUE THREAD

**Issue by dblock** [bbatsov/rubocop#2436]
Turns `Hash[{ x: 1 } => [1]]` into `Hash[{ x: 1 } => [1 }]` 

```ruby
describe RuboCop::Cop::Style::SpaceInsideHashLiteralBraces, :config do
  it 'accepts' do
    inspect_source(cop, 'Hash[{ x: 1 } => [1]]')
    expect(cop.offenses).to be_empty
  end

  it 'autocorrects' do
    new_source = autocorrect_source(cop, 'Hash[{ x: 1 } => [1]]')
    expect(new_source).to eq 'Hash[{ x: 1 } => [1]]'
  end
end
```

```
Failures:

  1) RuboCop::Cop::Style::SpaceInsideHashLiteralBraces autocorrects
     Failure/Error: expect(new_source).to eq 'Hash[{ x: 1 } => [1]]'
       
       expected: "Hash[{ x: 1 } => [1]]"
            got: "Hash[{ x: 1 } => [1 }]"
       
       (compared using ==)
     # ./spec/rubocop/cop/style/space_inside_hash_literal_braces_spec.rb:155:in `block (2 levels) in <top (required)>'

  2) RuboCop::Cop::Style::SpaceInsideHashLiteralBraces accepts
     Failure/Error: expect(cop.offenses).to be_empty
       expected `[#<RuboCop::Cop::Offense:0x007fcfe5ba1758 @severity=#<RuboCop::Cop::Severity:0x007fcfe5ba1730 @name=:convention>, @location=#<Parser::Source::Range (string) 19...20>, @message="Space inside ] missing.", @cop_name="Style/SpaceInsideHashLiteralBraces", @status=:uncorrected>].empty?` to return true, got false
     # ./spec/rubocop/cop/style/space_inside_hash_literal_braces_spec.rb:150:in `block (2 levels) in <top (required)>'
```

This is actual code in a spec in https://github.com/solnic/virtus
